packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart

```diff
@@ -287,6 +287,45 @@ void main() {
         SystemClock: () => clock,
         Usage: () => usage,
       });
+
+      testUsingContext('command release lock on kill signal', () async {
+        mockTimes = <int>[1000, 2000];
+        final Completer<void> completer = Completer<void>();
+        setExitFunctionForTests((int exitCode) {
+          expect(exitCode, 0);
+          restoreExitFunction();
+          completer.complete();
+        });
+        final DummyFlutterCommand flutterCommand = DummyFlutterCommand(
+            commandFunction: () async {
+              await Cache.lock();
+              final Completer<void> c = Completer<void>();
+              await c.future;
+              return null; // unreachable
+            }
+        );
+        unawaited(flutterCommand.run());
+        await Future<void>.delayed(const Duration(milliseconds: 50));
+        try {
+          await Cache.lock();
+        } on AssertionError catch (error) {
```

Comment:
Catching `AssertionError` is a little dicey since it won't be thrown if tests are run without assertions enabled. Can you rearrange this test around `Cache.checkLockAcquired()` instead? The `StateError` thrown there won't be subject to the same problem.

Reply:
Done, thank you